.tools/validate.go

```diff
@@ -0,0 +1,269 @@
+// An error should abort further tests.
+type ValidateRule func(CommitEntry) (ValidateResult, error)
+
+// ValidateCommit processes the given rules on the provided commit, and returns the result set.
+// An error should abort further tests.
+func ValidateCommit(c CommitEntry, rules []ValidateRule) (ValidateResults, error) {
+	results := ValidateResults{}
+	for _, r := range rules {
+		vr, err := r(c)
+		if err != nil {
+			return results, err
+		}
+		results = append(results, vr)
+	}
+	return results, nil
+}
+
+// ValidateResult is the result for a single validation of a commit.
+type ValidateResult struct {
+	CommitEntry CommitEntry
+	Pass        bool
+	Msg         string
+}
+
+// ValidateResults is a set of results. This is type makes it easy for the following function.
+type ValidateResults []ValidateResult
+
+// PassFail gives a quick over/under of passes and failures of the results in this set
+func (vr ValidateResults) PassFail() (pass int, fail int) {
+	for _, res := range vr {
+		if res.Pass {
+			pass++
+		} else {
+			fail++
+		}
+	}
+	return pass, fail
+}
+
+// CommitEntry represents a single commit's information from `git`
+type CommitEntry struct {
+	Commit               string
+	AbbreviatedCommit    string `json:"abbreviated_commit"`
+	Tree                 string
+	AbbreviatedTree      string `json:"abbreviated_tree"`
+	Parent               string
+	AbbreviatedParent    string `json:"abbreviated_parent"`
+	Refs                 string
+	Encoding             string
+	Subject              string
+	SanitizedSubjectLine string `json:"sanitized_subject_line"`
+	Body                 string
+	CommitNotes          string `json:"commit_notes"`
+	VerificationFlag     string `json:"verification_flag"`
+	ShortMsg             string
+	Signer               string
+	SignerKey            string       `json:"signer_key"`
+	Author               PersonAction `json:"author,omitempty"`
+	Commiter             PersonAction `json:"commiter,omitempty"`
+}
+
+// PersonAction is a time and identity of an action on a git commit
+type PersonAction struct {
+	Name  string `json:"name"`
+	Email string `json:"email"`
+	Date  string `json:"date"` // this could maybe be an actual time.Time
+}
+
+var (
+	prettyLogSubject       = `--pretty=format:%s`
+	prettyLogBody          = `--pretty=format:%b`
+	prettyLogCommit        = `--pretty=format:%H`
+	prettyLogAuthorName    = `--pretty=format:%aN`
+	prettyLogAuthorEmail   = `--pretty=format:%aE`
+	prettyLogCommiterName  = `--pretty=format:%cN`
+	prettyLogCommiterEmail = `--pretty=format:%cE`
+	prettyLogFormat        = `--pretty=format:{"commit": "%H", "abbreviated_commit": "%h", "tree": "%T", "abbreviated_tree": "%t", "parent": "%P", "abbreviated_parent": "%p", "refs": "%D", "encoding": "%e", "sanitized_subject_line": "%f", "commit_notes": "%N", "verification_flag": "%G?", "signer": "%GS", "signer_key": "%GK", "author": { "date": "%aD" }, "commiter": { "date": "%cD" }}`
```

Comment:
I'd drop `"commit_notes": "%N"`, since it seems like more user-generated content that could contain double quotes.  But I've never seen anyone actually use Git notes, so I'm not 100% sure that `%N` is the same stuff as `git notes add …`.  `"signer": "%GS"` should also be dropped from the JSON as user-generated content.